File: 01/2.py
floor = 0
i = 0
with open("input.txt", 'rb') as handle:
    while True:
        i += 1
        c = handle.read(1)
        if not c:
            print 'EOF', floor
            break
        if '(' == c:
            floor += 1
        elif ')' == c:
            floor += -1
            if floor < 0:
                print 'basement at', i
                break

File: 01/2.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"log"
	"os"
)

func main() {

	floor := 0
	counter := 0

	fi, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer func() {
		if err := fi.Close(); err != nil {
			panic(err)
		}
	}()

	r := bufio.NewReader(fi)
	for {
		if c, _, err := r.ReadRune(); err != nil {
			if err == io.EOF {
				break
			} else {
				log.Fatal(err)
			}
		} else {
			counter++
			if c == '(' {
				floor++
			} else if c == ')' {
				floor--
				if floor < 0 {
					break
				}
			}
		}
	}

	fmt.Printf("Basement at: %d\n", counter)

}
